src/Avalonia.Controls/TabControl.cs

```diff
@@ -166,10 +168,24 @@ protected override void OnPointerPressed(PointerPressedEventArgs e)
         {
             base.OnPointerPressed(e);
 
-            if (e.MouseButton == MouseButton.Left)
+            if (e.MouseButton == MouseButton.Left && e.Pointer.Type == PointerType.Mouse)
             {
                 e.Handled = UpdateSelectionFromEventSource(e.Source);
             }
         }
+
+        protected override void OnPointerReleased(PointerReleasedEventArgs e)
```

Comment:
Note: mouse release behavior is needed to allow scroll gesture recognizer to take over before TabControl switches the tab